vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/util/ExpressionToolbox.java

```diff
@@ -197,8 +197,11 @@ public static void getConstantAsPointable(ConstantExpression typeExpression, Tag
     }
 
     public static int getTypeExpressionTypeArgument(Mutable<ILogicalExpression> searchM) {
-        final int ARG_TYPE = 1;
+        int ARG_TYPE = 1;
         AbstractFunctionCallExpression searchFunction = (AbstractFunctionCallExpression) searchM.getValue();
+        if (searchFunction.getFunctionIdentifier().equals(BuiltinFunctions.FN_COUNT_1.getFunctionIdentifier())) {
```

Comment:
It was just a change that I made when I was trying to fix another rule and I forgot to remove it. Now, I deleted these lines in my new pull request.